src/shogun/machine/gp/SparseInference.h

```diff
@@ -303,6 +303,10 @@ class SparseInference: public Inference
 	/** inducing features for approximation */
 	SGMatrix<float64_t> m_inducing_features;
 
+	/**inducing features for approximation, should be same as
+	 * m_inducing_features*/
+	std::shared_ptr<Features> _m_inducing_features;
```

Comment:
why do we need this on top of the matrix?

Reply:
in constructor, inducing_features is Features
https://github.com/shogun-toolbox/shogun/blob/5a8edb65af72f49c1c7e061746b86005cc9113de/src/shogun/machine/gp/SparseInference.cpp#L92-L98
i think it's good that when we use put, inducting_features is also a Features

Reply:
why not only store the object and not the matrix? Could be a (private) getter instead for the matrix?